configure.ac

```diff
@@ -421,16 +425,23 @@ AS_IF([test "x$with_readline" != xno],[
 
   dnl We check for the symbol rl_ding to distinguish GNU readline
   dnl from other readline implementation.
-  AX_CHECK_LIBRARY([READLINE], [readline/readline.h], [readline], [rl_ding], [], [], [])
-
+  case "$ac_cv_search_rl_bind_keyseq" in
+     *ereadline)
+       AX_CHECK_LIBRARY([READLINE], [ereadline/readline/readline.h], [ereadline], [rl_ding], [], [], [])
+       ;;
+     *)
+       AX_CHECK_LIBRARY([READLINE], [readline/readline.h], [readline],  [rl_ding], [], [], [])
```

Comment:
I don't really like that now we are mixing `AC_SEARCH_LIBS` and `AX_CHECK_LIBRARY`, that kind of defeats the purpose of the latter, and also adds a redundant linker check (though perhaps if you always check for `rl_ding`, resp. always check for `rl_bind_keyseq`, autoconf's caching might take care of that...).

One alternative would be to simply use `AC_CHECK_HEADER` here, as we already performed the linker check.

Reply:
I suppose one can simply get rid of these calls, and do everything needed (calling `AC_DEFINE` and setting variables) within the scope of the `AC_SEARCH_LIBS` call.
I can do this change if you like.

Reply:
Yes, that sounds better. If you were willing to do that, would be great!

Reply:
It appears the check for `rl_ding` (appeared in readline 4.2) is strictly weaker than the check for  `rl_bind_keyseq` (appeared in readline 5.0). So I'll just check for headers, instead of calling  `AC_SEARCH_LIBS` or `AX_CHECK_LIBRARY`.

Reply:
We overlooked that due to non-standard header location on OpenBSD one should add
`-I/usr/local/include/ereadline` to `READLINE_CPPFLAGS` for this case. 
And also make sure that `AC_CHECK_HEADER` picks the right place by adding `READLINE_CPPFLAGS` to `CFLAGS` before calling it.

Reply:
done